docs(withings): add webhook troubleshooting and entity visibility guidance (#41759)
home-assistant/home-assistant.github.io: source/_integrations/withings.markdown

The {% term integration %} automatically detects if you can use webhooks. This enables the {% term integration %} only to update when there is new data.
The binary sensor for sleep will only work if the {% term integration %} can establish webhooks with Withings.

{% note %}
Sleep sensors will only appear when the integration can successfully receive webhook updates from Withings and find sleep data within the last day.
{% endnote %}

### Webhook requirements

For webhooks to work, your Home Assistant instance must be reachable by the Withings cloud service. The following requirements must be met:

1. Your Home Assistant instance must be reachable from the internet.
2. Withings requires HTTPS specifically on port 443. Using HTTPS on a non-standard port (such as 8443) will not work.

> 3. The certificate must be signed by a globally recognized Certificate Authority (for example, Let's Encrypt). Self-signed certificates will not work.

Reviewer: ```suggestion
- Your Home Assistant instance must be reachable from the internet.
- Withings requires HTTPS specifically on port 443. Using HTTPS on a non-standard port (such as 8443) will not work.
- The certificate must be signed by a globally recognized Certificate Authority (for example, Let's Encrypt). Self-signed certificates will not work.
```

The {% term integration %} automatically detects if you can use webhooks. This enables the {% term integration %} only to update when there is new data.
The binary sensor for sleep will only work if the {% term integration %} can establish webhooks with Withings.

> 

Reviewer: ```suggestion
```

The {% term integration %} automatically detects if you can use webhooks. This enables the {% term integration %} only to update when there is new data.
The binary sensor for sleep will only work if the {% term integration %} can establish webhooks with Withings.

> ### Webhook detection and troubleshooting

Reviewer 2: This belong in the troubleshooting section.

> ### Webhook detection and troubleshooting

Reviewer: ```suggestion
```

If you're having issues with webhook detection or your Withings Sleep Mat "in bed" entity is not showing up, please follow these troubleshooting steps:

1. **Verify webhook status**: Check your Home Assistant logs for messages about webhook registration. Successful webhook registration will show a message like "Successfully registered Withings webhook".
2. **Network accessibility**: For webhooks to work, your Home Assistant instance must be accessible from the internet. This requires:
   - A public IP address 
   - Port forwarding configured on your router (typically port 443 or 8123)
   - A domain name pointing to your public IP or using a dynamic DNS service (like DuckDNS)
   - SSL certificate configured (Let's Encrypt or a trusted certificate)
3. **Check Withings cloud connectivity**: If webhook registration fails, the integration will fall back to polling, which only updates every 5 minutes and doesn't support sleep sensors.

> 4. **Enable debug logging**: To see detailed information about webhook registration, enable debug logging by adding the following to your `configuration.yaml`:

Reviewer 2: This seems unneeded, users can enable debugging in the UI?

> 4. **Enable debug logging**: To see detailed information about webhook registration, enable debug logging by adding the following to your `configuration.yaml`:

Reviewer: ```suggestion
```

   
   ```yaml
   logger:
     default: warning
     logs:
       homeassistant.components.withings: debug

>    ```

Reviewer: ```suggestion
```

Workout calendar and the workout and activity sensors show if the latest available data point is no older than 14 days.

## Troubleshooting

If you're having issues with your Withings Sleep Mat or other devices not appearing, follow these steps:

### Sleep sensors not appearing

#### Symptom

Your Withings Sleep Mat entities are not showing up.

#### Description

Sleep sensors (including the "in bed" binary sensor) require successful webhook registration. 

#### Resolution

Check if webhooks are working properly as described in the [data updates](#data-updates) section above.

> - **In bed entity missing**: This entity specifically requires webhook functionality to work. If it's not appearing, ensure your Home Assistant instance is accessible from the internet and webhooks are properly registered.

Reviewer: ```suggestion
### In bed entity missing

#### Symptom

The **In bed** entity does not show up.

#### Description

This entity specifically requires webhook functionality to work. 

#### Resolution

Ensure your Home Assistant instance is accessible from the internet and webhooks are properly registered.
```

Workout calendar and the workout and activity sensors show if the latest available data point is no older than 14 days.

## Troubleshooting

If you're having issues with your Withings Sleep Mat or other devices not appearing, follow these steps:

### Sleep sensors not appearing

#### Symptom

Your Withings Sleep Mat entities are not showing up.

#### Description

Sleep sensors (including the "in bed" binary sensor) require successful webhook registration. 

#### Resolution

> Check if webhooks are working properly as described in the [data updates](#data-updates) section above.

Reviewer: ```suggestion
Check if webhooks are working properly as described in the [data updates](#data-updates) section.

```

### In bed entity missing

#### Symptom

The **In bed** entity does not show up.

#### Description

This entity specifically requires webhook functionality to work. 

#### Resolution

> Ensure your Home Assistant instance is accessible from the internet and webhooks are properly registered.

Reviewer: ```suggestion
- Ensure your Home Assistant instance is accessible from the internet and webhooks are properly registered.
```

> - **Check logs**: Enable debug logging to see webhook registration status as described in the Data Updates section.

Reviewer: ```suggestion
- [Enable debug logging](https://www.home-assistant.io/docs/configuration/troubleshooting/#debug-logs-and-diagnostics) to see webhook registration status as described in the [data updates](#data-updates) section.
```

Workout calendar and the workout and activity sensors show if the latest available data point is no older than 14 days.

## Troubleshooting

If you're having issues with your Withings Sleep Mat or other devices not appearing, follow these steps:

### Sleep sensors not appearing

#### Symptom

Your Withings Sleep Mat entities are not showing up.

#### Description

Sleep sensors (including the "in bed" binary sensor) require successful webhook registration. 

#### Resolution

Check if webhooks are working properly as described in the [data updates](#data-updates) section.

### In bed entity missing

#### Symptom

The **In bed** entity does not show up.

#### Description

This entity specifically requires webhook functionality to work. 

#### Resolution

- Ensure your Home Assistant instance is accessible from the internet and webhooks are properly registered.

> - [Enable debug logging](https://www.home-assistant.io/docs/configuration/troubleshooting/#debug-logs-and-diagnostics) to see webhook registration status as described in the [data updates](#data-updates) section.

Reviewer: ```suggestion
- [Enable debug logging](/docs/configuration/troubleshooting/#debug-logs-and-diagnostics) to see webhook registration status as described in the [data updates](#data-updates) section.
```

> ### General Troubleshooting Steps

Reviewer: ```suggestion
### General troubleshooting steps

```

1. Make sure your Withings device is properly set up in the Withings app and showing data
2. Verify that data exists in your Withings account from within the last 14 days (or last day for sleep data)
3. Wait up to 15 minutes after adding the integration for all entities to appear
4. If using a sleep mat, ensure it has recorded sleep data within the last day

> 5. Restart the integration by removing and re-adding it if entities still don't appear after verifying the above

Reviewer: ```suggestion
1. Make sure your Withings device is properly set up in the Withings app and showing data.
2. Verify that data exists in your Withings account from within the last 14 days (or last day for sleep data).
3. Wait up to 15 minutes after adding the integration for all entities to appear.
4. If using a sleep mat, ensure it has recorded sleep data within the last day.
5. Restart the integration by removing and re-adding it if entities still don't appear after verifying the above.
```

{% note %}
If you're still having issues after following these steps, enable debug logging to gather more information about what might be preventing the integration from working properly.
{% endnote %}

### Webhook detection and troubleshooting

If you're having issues with webhook detection or your Withings Sleep Mat "in bed" entity is not showing up, please follow these troubleshooting steps:

1. **Verify webhook status**: Check your Home Assistant logs for messages about webhook registration. Successful webhook registration will show a message like "Successfully registered Withings webhook".
2. **Network accessibility**: For webhooks to work, your Home Assistant instance must be accessible from the internet. This requires:
   - A public IP address 
   - Port forwarding configured on your router (typically port 443 or 8123)
   - A domain name pointing to your public IP or using a dynamic DNS service (like DuckDNS)
   - SSL certificate configured (Let's Encrypt or a trusted certificate)
3. **Check Withings cloud connectivity**: If webhook registration fails, the integration will fall back to polling, which only updates every 5 minutes and doesn't support sleep sensors.

> 4. **Enable debug logging**: To see detailed information about webhook registration, enable debug logging

Reviewer: ```suggestion
1. Verify webhook status: Check your Home Assistant logs for messages about webhook registration. Successful webhook registration will show a message like "Successfully registered Withings webhook".
2. Check network accessibility: For webhooks to work, your Home Assistant instance must be accessible from the internet. This requires the following:
   - A public IP address 
   - Port forwarding configured on your router (typically port 443 or 8123)
   - A domain name pointing to your public IP or using a dynamic DNS service (like DuckDNS)
   - SSL certificate configured (Let's Encrypt or a trusted certificate)
3. Check Withings cloud connectivity: If webhook registration fails, the integration will fall back to polling, which only updates every 5 minutes and doesn't support sleep sensors.
4. Enable debug logging: To see detailed information about webhook registration,  [enable debug logging](/docs/configuration/troubleshooting/#debug-logs-and-diagnostics).
```

{% note %}
If you're still having issues after following these steps, enable debug logging to gather more information about what might be preventing the integration from working properly.
{% endnote %}

### Webhook detection and troubleshooting

If you're having issues with webhook detection or your Withings Sleep Mat "in bed" entity is not showing up, please follow these troubleshooting steps:

1. Verify webhook status: Check your Home Assistant logs for messages about webhook registration. Successful webhook registration will show a message like "Successfully registered Withings webhook".
2. Check network accessibility: For webhooks to work, your Home Assistant instance must be accessible from the internet. This requires the following:
   - A public IP address 
   - Port forwarding configured on your router (typically port 443 or 8123)
   - A domain name pointing to your public IP or using a dynamic DNS service (like DuckDNS)
   - SSL certificate configured (Let's Encrypt or a trusted certificate)
3. Check Withings cloud connectivity: If webhook registration fails, the integration will fall back to polling, which only updates every 5 minutes and doesn't support sleep sensors.

> 4. Enable debug logging: To see detailed information about webhook registration,  [enable debug logging](/docs/configuration/troubleshooting/#debug-logs-and-diagnostics).

Reviewer: ```suggestion
4. Enable debug logging: To see detailed information about webhook registration,  [enable debug logging](/docs/configuration/troubleshooting/#debug-logs-and-diagnostics).

```

Workout calendar and the workout and activity sensors show if the latest available data point is no older than 14 days.

## Troubleshooting

If you're having issues with your Withings Sleep Mat or other devices not appearing, follow these steps:

> ### Webhook and Sleep Sensor Issues

Reviewer: ```suggestion
```

> - **Sleep sensors not appearing**: Sleep sensors (including the "in bed" binary sensor) require successful webhook registration. If your Withings Sleep Mat entities are not showing up, check if webhooks are working properly as described in the Data Updates section above.

Reviewer: ```suggestion
### Sleep sensors not appearing

#### Symptom

Your Withings Sleep Mat entities are not showing up.

#### Description

Sleep sensors (including the "in bed" binary sensor) require successful webhook registration. 

#### Resolution

Check if webhooks are working properly as described in the [data updates](#data-updates) section above.
```

#### Symptom

Your Withings Sleep Mat entities are not showing up.

#### Description

Sleep sensors (including the "in bed" binary sensor) require successful webhook registration. 

#### Resolution

Check if webhooks are working properly as described in the [data updates](#data-updates) section.

### In bed entity missing

#### Symptom

The **In bed** entity does not show up.

#### Description

This entity specifically requires webhook functionality to work. 

#### Resolution

- Ensure your Home Assistant instance is accessible from the internet and webhooks are properly registered.
- [Enable debug logging](/docs/configuration/troubleshooting/#debug-logs-and-diagnostics) to see webhook registration status as described in the [data updates](#data-updates) section.

### General troubleshooting steps

1. Make sure your Withings device is properly set up in the Withings app and showing data.
2. Verify that data exists in your Withings account from within the last 14 days (or last day for sleep data).
3. Wait up to 15 minutes after adding the integration for all entities to appear.
4. If using a sleep mat, ensure it has recorded sleep data within the last day.

> 5. Restart the integration by removing and re-adding it if entities still don't appear after verifying the above.

Reviewer 3: Is it really required to remove the config entry and add it again? Is reloading the config entry not enough?

- **In bed entity missing**: This entity specifically requires webhook functionality to work. If it's not appearing, ensure your Home Assistant instance is accessible from the internet and webhooks are properly registered.
- **Check logs**: Enable debug logging to see webhook registration status as described in the Data Updates section.

### General Troubleshooting Steps
1. Make sure your Withings device is properly set up in the Withings app and showing data
2. Verify that data exists in your Withings account from within the last 14 days (or last day for sleep data)
3. Wait up to 15 minutes after adding the integration for all entities to appear
4. If using a sleep mat, ensure it has recorded sleep data within the last day
5. Restart the integration by removing and re-adding it if entities still don't appear after verifying the above

{% note %}
If you're still having issues after following these steps, enable debug logging to gather more information about what might be preventing the integration from working properly.
{% endnote %}

> 

Reviewer: ```suggestion

### Webhook detection and troubleshooting

If you're having issues with webhook detection or your Withings Sleep Mat "in bed" entity is not showing up, please follow these troubleshooting steps:

1. **Verify webhook status**: Check your Home Assistant logs for messages about webhook registration. Successful webhook registration will show a message like "Successfully registered Withings webhook".
2. **Network accessibility**: For webhooks to work, your Home Assistant instance must be accessible from the internet. This requires:
   - A public IP address 
   - Port forwarding configured on your router (typically port 443 or 8123)
   - A domain name pointing to your public IP or using a dynamic DNS service (like DuckDNS)
   - SSL certificate configured (Let's Encrypt or a trusted certificate)
3. **Check Withings cloud connectivity**: If webhook registration fails, the integration will fall back to polling, which only updates every 5 minutes and doesn't support sleep sensors.
4. **Enable debug logging**: To see detailed information about webhook registration, enable debug logging
```
